tests/test_datasets/test_hierarchical_structure.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict
+from typing import List
+
+import numpy as np
+import pytest
+
+from etna.datasets import HierarchicalStructure
+
+
+@pytest.fixture
+def simple_hierarchical_struct():
+    return HierarchicalStructure({"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d"]}, ["l1", "l2", "l3"])
+
+
+@pytest.fixture
+def tailed_hierarchical_struct():
+    return HierarchicalStructure(
+        {"total": ["X", "Y"], "X": ["a"], "Y": ["c", "d"], "c": ["f"], "d": ["g"], "a": ["e", "h"]},
+        ["l1", "l2", "l3", "l4"],
+    )
+
+
+@pytest.mark.parametrize(
+    "target,source,answer",
+    (
+        ("l1", "l2", np.array([[1, 1]])),
+        ("l1", "l4", np.array([[1, 1, 1, 1]])),
+        ("l2", "l3", np.array([[1, 0, 0], [0, 1, 1]])),
+    ),
+)
+def test_tailed_struct_matrix(
+    tailed_hierarchical_struct: HierarchicalStructure, source: str, target: str, answer: np.ndarray
+):
+    np.testing.assert_array_almost_equal(
+        answer, tailed_hierarchical_struct.get_summing_matrix(target, source).toarray()
+    )
+
+
+@pytest.mark.parametrize(
+    "target,source,answer",
+    (
+        ("l1", "l2", np.array([[1, 1]])),
+        ("l1", "l3", np.array([[1, 1, 1, 1]])),
+        ("l2", "l3", np.array([[1, 1, 0, 0], [0, 0, 1, 1]])),
+    ),
+)
+def test_simple_struct_matrix(
+    simple_hierarchical_struct: HierarchicalStructure, source: str, target: str, answer: np.ndarray
+):
+    np.testing.assert_array_almost_equal(
+        answer, simple_hierarchical_struct.get_summing_matrix(target, source).toarray()
+    )
+
+
+@pytest.mark.parametrize(
+    "structure",
+    (
+        {"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d", "total"]},
+        {"X": ["a", "b"], "Y": ["c", "d"]},
+        dict(),
+    ),
+)
+def test_root_finding_errors(structure: Dict[str, List[str]]):
+    with pytest.raises(ValueError, match="Invalid tree definition: unable to find root!"):
+        HierarchicalStructure(structure)
+
+
+@pytest.mark.parametrize(
+    "structure",
+    (
+        {"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d"], "a": ["X"]},
```

Comment:
Add short comments to the cases, just to make it easy to check that you cover all possible cases(multiple components, loops, different depth of leafs). You can parametrize it with error msg also